huaweicloud/services/dli/resource_huaweicloud_dli_datasource_auth.go

```diff
@@ -0,0 +1,352 @@
+			201,
+		},
+	}
+	createDatasourceAuthOpt.JSONBody = utils.RemoveNil(buildCreateDatasourceAuthBodyParams(d, cfg))
+	_, err = createDatasourceAuthClient.Request("POST", createDatasourceAuthPath, &createDatasourceAuthOpt)
+	if err != nil {
+		return diag.Errorf("error creating DatasourceAuth: %s", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+
+	return resourceDatasourceAuthRead(ctx, d, meta)
+}
+
+func buildCreateDatasourceAuthBodyParams(d *schema.ResourceData, _ *config.Config) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"auth_info_name":       utils.ValueIngoreEmpty(d.Get("name")),
+		"datasource_type":      utils.ValueIngoreEmpty(d.Get("type")),
+		"username":             utils.ValueIngoreEmpty(d.Get("username")),
+		"password":             utils.ValueIngoreEmpty(d.Get("password")),
+		"certificate_location": utils.ValueIngoreEmpty(d.Get("certificate_location")),
+		"truststore_location":  utils.ValueIngoreEmpty(d.Get("truststore_location")),
+		"truststore_password":  utils.ValueIngoreEmpty(d.Get("truststore_password")),
+		"keystore_location":    utils.ValueIngoreEmpty(d.Get("keystore_location")),
+		"keystore_password":    utils.ValueIngoreEmpty(d.Get("keystore_password")),
+		"key_password":         utils.ValueIngoreEmpty(d.Get("key_password")),
+		"krb5_conf":            utils.ValueIngoreEmpty(d.Get("krb5_conf")),
+		"keytab":               utils.ValueIngoreEmpty(d.Get("keytab")),
+	}
+	return bodyParams
+}
+
+func resourceDatasourceAuthRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getDatasourceAuth: Query the DLI datasource authentication.
+	var (
+		getDatasourceAuthHttpUrl = "v2.0/{project_id}/datasource/auth-infos"
+		getDatasourceAuthProduct = "dli"
+	)
+	getDatasourceAuthClient, err := cfg.NewServiceClient(getDatasourceAuthProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DLI Client: %s", err)
+	}
+
+	getDatasourceAuthPath := getDatasourceAuthClient.Endpoint + getDatasourceAuthHttpUrl
+	getDatasourceAuthPath = strings.ReplaceAll(getDatasourceAuthPath, "{project_id}", getDatasourceAuthClient.ProjectID)
+
+	getDatasourceAuthqueryParams := buildGetDatasourceAuthQueryParams(d)
+	getDatasourceAuthPath += getDatasourceAuthqueryParams
+
+	getDatasourceAuthOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	getDatasourceAuthResp, err := getDatasourceAuthClient.Request("GET", getDatasourceAuthPath, &getDatasourceAuthOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving DatasourceAuth")
+	}
+
+	getDatasourceAuthRespBody, err := utils.FlattenResponse(getDatasourceAuthResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("name", utils.PathSearch("auth_infos[0].auth_info_name", getDatasourceAuthRespBody, nil)),
```

Comment:
I think you can save the "auth_infos[0]", and check whether it exist

Reply:
ok